shared/settings.cpp

```diff
@@ -0,0 +1,102 @@
+#include <stdlib.h>
+#include <string.h>
+#include <stdio.h>
+#include <QDebug>
+#include "settings.h"
+
+char* readFile(const char* path) {
+    FILE *file = fopen(path, "r");
+    if (file) {
+        fseek(file, 0, SEEK_END);
+        long length=ftell(file);
+        fseek (file, 0, SEEK_SET);
+        char *buffer = (char*)malloc(length+1);
+        if (buffer) {
+            fread (buffer, 1, length, file);
+            buffer[length] = '\0';
+        }
+        fclose(file);
+        return buffer;
+    }
+    return nullptr;
+}
+
+bool doesStartWithPrefixes(const char* const prefixes[], int arrLength, char* toCheck) {
+    int length = strlen(toCheck);
+    for (int i = 0; i < arrLength; ++i) {
+        int prefixLength = strlen(prefixes[i]);
+        if (prefixLength < length && strncmp(prefixes[i], toCheck, prefixLength) == 0) {
+            return true;
+        }
+    }
+    return false;
+}
+
+void Settings::readDeviceType() {
+    char* device_type = readFile("/sys/devices/soc0/machine");
```

Comment:
This can be simplified by using QFile to read the contents, and QString also has helpers for string comparison.

Reply:
You can see a sample of QFile usage here: https://github.com/Eeems/oxide/blob/64b2b689b4562bd84fa0eeed67739dc9465a9a83/applications/system-service/sysobject.cpp#L28-L41